pmemkv/src/main/java/site/ycsb/db/pmemkv/PmemKVClient.java

```diff
@@ -119,31 +121,60 @@ public void init() throws DBException {
         String engineName = props.getProperty(ENGINE_PROPERTY, "cmap");
 
         String path = props.getProperty(PATH_PROPERTY);
-        if (path == null) {
-          throw new DBException(PATH_PROPERTY + " is obligatory to run PmemKV client");
-        }
         String size = props.getProperty(SIZE_PROPERTY);
-        if (size == null) {
-          throw new DBException(SIZE_PROPERTY + " is obligatory to run PmemKV client");
+        String jsonConfigProp = props.getProperty(JSON_CONFIG_PROPERTY);
+        Path jsonConfigFile = null;
+        String jsonConfigContent = null;
+        if (jsonConfigProp != null && !jsonConfigProp.isEmpty()) {
+          jsonConfigFile = Paths.get(jsonConfigProp);
+          try {
+            if (jsonConfigFile != null) {
+              jsonConfigContent = new String(Files.readAllBytes(jsonConfigFile), UTF_8);
+            }
+          } catch (IOException e) {
+            throw new DBException(JSON_CONFIG_PROPERTY +
+              " contain faulty path or points to improper json config params");
+          }
         }
+
         boolean startError = false;
+        Database.Builder<byte[], Map<String, ByteIterator>> builder = null;
+
+        /* try to open db first */
         try {
-          /* try to open db first */
-          db = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName)
-              .setSize(Long.parseLong(size))
-              .setPath(path)
+          builder = new Database.Builder<byte[], Map<String, ByteIterator>>(engineName);
+          if (jsonConfigContent != null) {
```

Comment:
It wouldn't be `null`, because code with string assignment was successful.
`jsonConfigContent != null` will be always true.
So please change it to:
`!jsonConfigContent.isEmpty()`
We have reference to created string object in 132 line, but we don't know this file was empty (it's an example).

Reply:
that's not true - we set `jsonConfigContent` only if parameter `jsonConfigProp` is given.

as for empty string - the content will be parsed by our pmemkv_json helper library and an error will be returned.

Reply:
Okey, I get it